pkg/kubelet/eviction/helpers.go

```diff
@@ -586,27 +588,59 @@ func (ms *multiSorter) Less(i, j int) bool {
 	return ms.cmp[k](p1, p2) < 0
 }
 
-// qosComparator compares pods by QoS (BestEffort < Burstable < Guaranteed)
-func qosComparator(p1, p2 *v1.Pod) int {
-	qosP1 := v1qos.GetPodQOS(p1)
-	qosP2 := v1qos.GetPodQOS(p2)
-	// its a tie
-	if qosP1 == qosP2 {
+// priority compares pods by Priority, if priority is enabled.
+func priority(p1, p2 *v1.Pod) int {
```

Comment:
The behavior change is intentional.  Without priority enabled, the behavior is very similar to old behavior, but with a small difference.  The new behavior mimics ranking by QoS since Guaranteed pods always have usage < requests, and best-effort pods always have usage > requests.  The only different scenario is that a burstable pod consuming far above requests will be evicted before a best-effort pod that does not consume much.

Reply:
Ok, I am ok with this small behavior changes on this. @derekwaynecarr any concerns?